### ISSUE TITLE
[BUG] Incorrect statistics values

### ISSUE THREAD

**Issue by bcnewlin** [OpenSIPS/opensips#2460]
**OpenSIPS version you are running**

Fails on version:
```
version: opensips 2.4.9 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, SHM_MMAP, PKG_MALLOC, F_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: b9b2bf9d1
main.c compiled on 00:49:28 Apr  8 2021 with gcc 7
```
Successful on version:
```
version: opensips 2.4.9 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, SHM_MMAP, PKG_MALLOC, F_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: 3bed2f312
main.c compiled on 02:35:25 Apr  8 2021 with gcc 7
```

**Describe the bug**

**To Reproduce**

1. Start OpenSIPS
2. Place a few calls
3. Notice the "unsupported_methods" value is incorrect

**Expected behavior**

Statistics should display accurate counts.

**Relevant System Logs**

Failing logs: https://pastebin.com/pDadw84F
Success logs: https://pastebin.com/c0Nj5b3A

**OS/environment information**
 - Operating System:  amazonlinux:2 on Docker
 - OpenSIPS installation:  git
 - other relevant information:

**Additional context**

**Comment by razvancrainea** [comment]
Hi, Ben!

Unfortunately the `unsupported_methods` statistics is quite broken by design. This statistic is incremented every time OpenSIPS tries to parse an unknown method, but the problem is that, for example, the method might be parsed several times for a request - once to get the request method, other to get the CSeq method, whatever - that would increment the `unsupported_methods` statistics counter several times for the same request, which is probably not what you want.
Besides that, it is a bit unclear what an `unsupported_method` means, because probably the more useful statistic you would want would be the `unhandled_method` - methods that your script does not know how to handle.
The actual reason of these changes between the 2.4 versions is the f2b46975ede352a9ab5ee88fd10351f610e4f331 commit, which tries to unify the "faked/dummy" messages in a single `DUMMY` message, that is being used in places where a request is not available (such as events unrelated to requests). This faked message gets to be parsed somewhere in the code, thus incrementing the `unsupported_methods`. A quick fix would be to consider `DUMMY` one of the known statistics, thus, not updating the statistic. However, due to the way the statistic itself is implemented, I would say it is more or less useless, and I'd suggest removing it. You can always (and more accurate) create your own statistic in the script, and increment it for all unknown methods you are handling.
Let us know what you think.

Best regards,
Răzvan

**Comment by bcnewlin** [comment]
@razvancrainea Thank you for that detailed explanation. We were not using this statistic for anything speific, but our testing is configured to verify all statistics for no unexpected values, so our tests were failing when this statistic was >0 for test where no unsupported SIP methods were present. If the statistic is unreliable we can remove that verification, but I wanted to be sure it wasn't a larger issue with statistics in general.

The only thing I would suggest is that if this statistic is truly not helpful, especially due to the changes around dummy messages, that maybe it should just be removed? I noticed it is still documented in `devel` version, though I didn't not verify it actually exists. At the very least, it seems updating the documentation to provide some of the information you gave here would be helpful, even if only to prevent anyone else from opening a ticket on the behavior.

**Comment by bogdan-iancu** [comment]
I would strongly suggest to drop this statistic. Even before the "DUMMY" request change, the stat was misleading - it counts requests which are not officially recognized by the parser (still accepted), far from being "unsupported". OpenSIPS will be fine in proxying and replying to a "BOGDAN" SIP method if the case :P .

**Comment by razvancrainea** [comment]
I've just removed the statistic from master (3.2) and in released versions (3.1 and 2.4) will no longer update it, thus always returning 0.
Thanks for your report!

**Comment by bcnewlin** [comment]
Thank you @razvancrainea !
